protocol/communities/manager.go

```diff
@@ -1881,14 +1881,49 @@ func (m *Manager) DeleteChat(communityID types.HexBytes, chatID string) (*Commun
 		return nil, nil, err
 	}
 
+	// Check for channel permissions
+	changes := community.emptyCommunityChanges()
+	for tokenPermissionID, tokenPermission := range community.tokenPermissions() {
+		currentChatIndex := -1
+		for i, chat := range tokenPermission.GetChatIds() {
+			if chat == chatID {
+				currentChatIndex = i
+				break
+			}
+		}
+		if currentChatIndex == -1 {
+			continue
+		}
+
+		if len(tokenPermission.GetChatIds()) == 1 {
+			// Delete channel permission, if there is only one channel
+			deletePermissionChanges, err := community.DeleteTokenPermission(tokenPermissionID)
+			if err != nil {
+				return nil, nil, err
+			}
+			changes.Merge(deletePermissionChanges)
+		} else {
+			// Remove the channel from the permission, if there are other channels
+			tokenPermission.ChatIds = append(tokenPermission.GetChatIds()[:currentChatIndex], tokenPermission.GetChatIds()[currentChatIndex+1:]...)
```

Comment:
_minor thing, I don't care much, decide on your own. I just hate these long slices concatenation and lookups 😄_

No idea why `chatIds` are `repeated string` and not a map. 
But imaging how much more enjoyable it could be!

```go
func (p *CommunityTokenPermission) ChatIdsMap() map[string]struct{}{} {
    chats := map[string]struct{}{}
    for _, id := range GetChatIds() {
        chats[id] = struct{}{}
    }
    return chats
}
```

```suggestion
        chats := tokenPermission.ChatIdsMap()
		chat, ok := chats[chatID]
		if !ok {
			continue
		}

		if len(tokenPermission.GetChatIds()) == 1 {
			// Delete channel permission, if there is only one channel
			deletePermissionChanges, err := community.DeleteTokenPermission(tokenPermissionID)
			if err != nil {
				return nil, nil, err
			}
			changes.Merge(deletePermissionChanges)
		} else {
			// Remove the channel from the permission, if there are other channels
			delete(chats, chatID)
			tokenPermission.ChatIds = maps.Keys(chats)
```

Reply:
I take your point. ill fix it now